### PR TITLE
Add an anonymous module to Mesos that modifies logging behavior.

### PR DESCRIPTION
**Pull request by kaysoky**
The prior change to logging (`tee`-ing stderr to a file) resulted in some incorrect agent recovery behavior.  When the agent exited or crashed (for any reason), there was a chance that the `tee` process would stay behind (presumably because the agent's stdout/stderr was leaked/inherited by a subprocess) indefinitely.  This led to the agent being removed and tasks being lost.

## This commit: 
* Removes the `tee_stderr` wrapper.
* Adds an anonymous module instead, which implements a `glog` `LogSink` that writes to a file.  Because this is a module, it is guaranteed to die if the master/agent dies too.
  * Updates the `mesos-modules` package.  https://github.com/dcos/dcos-mesos-modules/pull/3

## Checklist
* [x] ~Add an agent recovery integration test:~ This requires tests to have SSH access, rather than interacting with exposed endpoints like all the other tests.  As such, this test is left out for now.
  1) Launch a simple sleep
  2) Tell the agent to exit (directly, not via systemd)
  3) Check for agent recovery and task existence
* [x] Make sure the `LogSink` plays well with the `logrotate` service.  (Probably a manual test)
* [x] Merge the `dcos-mesos-modules` PR too.

### CODE REVIEW

**Comment by kaysoky** [comment]
Launched a test cluster and validated that the master and agent have an additional log file that contains `logsink.cpp` logs:
```
I1129 21:23:58.000000  1917 logsink.cpp:89] Added FileSink for glog logs to: /var/lib/dcos/mesos/log/mesos-master.log
```
```
I1129 21:25:06.000000  3253 logsink.cpp:89] Added FileSink for glog logs to: /var/log/mesos/mesos-agent.log
```

**Comment by kaysoky** [comment]
Spammed some tasks (and therefore log lines) with:
```
while true; do sh -c './mesos-execute --master=master.mesos:5050 --name="sleep" --command="sleep 0" &'; done
```
Verified that logging to the leading log file continues after rotation (copytruncate).

**Comment by kaysoky** [comment]
Ready for review!

I'll merge the `dcos-mesos-modules` part when we figure out the SHA of Mesos (#977)

File: packages/mesos-modules/buildinfo.json [hunk 1]
```diff
@@ -2,8 +2,8 @@
   "requires": ["mesos"],
     "single_source" : {
       "kind": "git",
-      "git": "https://github.com/dcos/dcos-mesos-modules.git",
-      "ref": "1e264ece1a69837d0b5da5e397852424f9b4a3eb",
+      "git": "https://github.com/kaysoky/dcos-mesos-modules.git",
```
Comment (cmaloney) [inline packages/mesos-modules/buildinfo.json]: note: needs to point to `dcos/dcos-mesos-modules` before could merge

**Comment by kaysoky** [comment]
Looks like #977 will bump the `dcos-mesos-modules` SHA to one including a ContainerLogger interface change.  Therefore, I can merge the anonymous module without any problems.

The SHA and reference branch in this PR have been updated.

**Comment by spahl** [comment]
cc @adam-mesos

**Comment by adam-mesos** [comment]
Looks good to me, but I guess you need the Mesos bump from #977 before this will actually build. You can pull it in yourself (rebase on the `adam/pods` branch) or we can try to land 977 now that it's all green (although there are EE changes required before we can bump the upstream with latest Mesos).

**Comment by cmaloney** [comment]
WIP because build fails

**Comment by adam-mesos** [comment]
Rebased now that Mesos and dcos-mesos-modules are already sufficiently bumped in dcos/dcos master.

**Comment by cmaloney** [comment]
@kaysoky could you rebase this on current master?

**Comment by kaysoky** [comment]
Rebased.  This PR no longer needs to change the `mesos-modules` package SHA as the existing SHA already contains the module.

**Comment by adam-mesos** [comment]
@cmaloney Could you give one final review here? After @kaysoky 's last rebase, there's no module-bump included in this PR anymore. I pulled this into https://github.com/dcos/dcos/pull/1131 which is based off master, and it applied cleanly. I'm ready to merge it (when Train 118 leaves) if you don't have any objections.

**Comment by sambatyon** [comment]
👍 Looks good to me!

**Comment by amitaekbote** [comment]
@kaysoky looks like the code quality is failing, can you fix that and we can pull this in the next train?

**Comment by adam-mesos** [comment]
That code quality error is old, and should be fixed/green if we just rebase this PR.
It was green when I pulled it into a previous train.

**Comment by kaysoky** [comment]
Yeah, this PR fixed it: https://github.com/dcos/dcos/pull/1114

**Pull request merged**